skunkworks_crow/src/main/java/org/odk/share/views/ui/main/MainActivity.java

```diff
@@ -213,9 +214,12 @@ private void showAlertDialog() {
 
         builder.setPositiveButton(getString(R.string.install), (DialogInterface dialog, int which) -> {
             try {
-                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + COLLECT_PACKAGE)));
+                startActivityForResult(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + COLLECT_PACKAGE)),
+                        COLLECT_INSTALL_REQUEST_CODE);
             } catch (ActivityNotFoundException e) {
-                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=" + COLLECT_PACKAGE)));
+                startActivityForResult(new Intent(Intent.ACTION_VIEW,
```

Comment:
What if this also results into an Exception, have you thought about that case?

Reply:
This exception handling was already there, I just changed the method to start an activity. Also, why would this result in an exception? As the browser is present in every android device, so it will definitely behave as intended.